docs/DeveloperGuide.md

```diff
@@ -1,34 +1,600 @@
+          Pros: Allow faster lookup of patients’ information using the unique identifier (nric)
+          Cons: Implementation is harder and may result in bugs if not implemented accurately.
+            
+        Alternative 2: Array List
+          Pros: This would be easier to implement and retrieve the information.
+          Cons: When a patient is deleted, all the patients in the patient list need to be checked. 
+                This would cause the deletion to be very slow when there is a large number of patients in the list.
+
+#### 4.1.1 Add Patient Details
+
+The user is able to add patients into the program to keep track of the patient's details. 
+The command: 
+
+    add patient /ic S7777777Z /n Alice /p 98765432 /d 01/01/2000 /b B+ /a Peanuts
+    
+will add a patient with `NRIC` as S7777777Z with the following attributes: 
+* name: `Alice`
+* phone number: `98765432`
+* date of birth: `01/01/2000`
+* blood type: `B+`
+* allergies: `Peanuts`
+
+**Implementation** 
+
+The activity diagram below summarises the process of executing an `add` command.
+![Add Patient Sequence Diagram](images/AddPatientSequenceDiagram.jpg)
+
+The following steps explains the sequence of events: 
+
+1. The user enters `add patient /ic S7777777Z /n Alice /p 98765432 /d 01/01/2000 /b B+ /a Peanuts`.
+
+2. `HappyPills` calls `Parser#parse()` which then calls `PatientParser#parse()`. 
+
+3. `PatientParser#parse()` will call its own method `PatientParser#parseAddCommand()`.
+
+4. `PatientParser#parseAddCommand()` will first validate the attributes and then create an object `AddPatientCommand` 
+if the attributes are valid. 
+
+5. `HappyPills` then calls `AddPatientCommand.execute()` to execute the command. 
+
+6. In `AddPatientCommand.execute()`, a patient is added and the display message is returned.
+
+#### 4.1.2 Edit Patient Details
+
+The user is able to edit the details of a patient in the list of patients currently in the program. The command: 
+
+    edit patient S7777777Z /a School
+    
+will edit the patient's allergies to `School`. 
+
+**Implementation** 
+
+![Edit Patient Sequence Diagram](images/EditPatientSequenceDiagram.jpg)
+
+The following steps explains the sequence of events: 
+
+1. The user enters `edit patient S7777777Z /a School` 
+
+2. `HappyPills` calls `Parser#parse()` which then calls `PatientParser#parse()`.
+
+3. `PatientParser#parse()` will then create an object `EditPatientCommand`. 
+
+4. `HappyPills` then calls `EditPatientCommand.execute()` method to execute the command. 
+
+5. In `EditPatientCommand.execute()`, the given attribute of the patient is edited and the display message is returned. 
+
+#### 4.1.3. Delete Patient Details 
+
+The user can delete a patient from the list of patients currently in the program. The command: 
+
+    delete patient S7777777Z 
+    
+will delete the patient with NRIC `S7777777Z`, if found. 
+
+**Implementation** 
+
+![Delete Patient Sequence Diagram](images/DeletePatientSequenceDiagram.jpg)
```

Comment:
![image](https://user-images.githubusercontent.com/15946120/78002271-8a767200-7369-11ea-98f9-b325a0b25dcd.png)

Typo error again and similar sentiments as previous diagrams